docs/rule/no-action-on-submit-button.md

```diff
@@ -0,0 +1,35 @@
+# no-action-on-submit-button
+
+This rule requires all `<button>` elements with a `type="submit"` attribute to not have any click action.
+
+When the `type` attribute of `<button>` elements is `submit`, the action should be on the `<form>` element instead of directly on the button.
+
+By default, the `type` attribute of `<button>` elements is `submit`.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<button type="submit" {{on "click" this.handleClick}} />
+<button type="submit" {{action "handleClick"}} />
+<button {{on "click" this.handleClick}} />
+<button {{action "handleClick"}} />
+```
+
+This rule **allows** the following:
+
+```hbs
+<button type="button" {{on "click" this.handleClick}} />
+<button type="button" {{action "handleClick"}} />
+<button type="submit" />
+<button />
```

Comment:
This would be disallowed by the `require-button-type` rule.

Reply:
Thanks for your comment @MelSumner. Sure, but aren't rules independent from each other?

My point is: should we take an other rule into account in the unit test of this current rule?

If users decide to disable the `require-button-type` rule, then this syntax will be allowed by the current rule, because it's not its role to prevent the missing type.

What is your opinion about it?

Reply:
So I try to write new rules with existing rules in mind; I don't want a rule I create to fail another existing rule. 

But @rwjblue might have some more clarity here. 

(PS overall I am excited about this rule!)

Reply:
Hi @MelSumner 

I'm back at work on this rule, which I'd forgotten for a long time 🙇 

This rule is not encouraging `<button />` without (explicit) type.  It's essentially a syntax that won't make this rule fail, so I listed it in the "valid" scenarios.

Would you prefer me to remove `<button />` from these scenarios?

- The downside is that this scenario won't be clearly mentioned while likely to happen.
- The advantage is that it won't encourage people using this syntax (not allowed by `require-button-type` dedicated rule)